<launch>

  <!-- specify the planning pipeline -->
  <arg name="pipeline" default="ompl" />

  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false" />

  <!-- Parameters for the module names, gains, hrdf, etc -->
  <!-- Note: the "moveit_joints" must matches the configuration from the MoveIt config -->
  <rosparam>
    names: [ "Base", "Shoulder", "Elbow", "Wrist1", "Wrist2" ]
    moveit_joints: ["HEBI/base/X5_4", "HEBI/shoulder/X5_9", "HEBI/elbow/X5_4", "HEBI/wrist1/X5_1"]
    families: [ "HEBI" ]
    gains_package: "hebi_cpp_api_examples"
    gains_file: "gains/4_dof_arm_gains.xml"
    hrdf_package: "hebi_cpp_api_examples"
    hrdf_file: "hrdf/4_dof_arm.hrdf"
    home_position: [ 0.01, 2.09439, 2.09439, 0.01 ]
  </rosparam>

  <node pkg="hebi_cpp_api_examples" type="moveit_arm_node" name="moveit_arm_node" output="screen" required="true"/>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(find hebi_a-2085-04_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- Run the main MoveIt! executable -->
  <include file="$(find hebi_a-2085-04_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="pipeline" value="$(arg pipeline)"/>
  </include>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <include file="$(find hebi_a-2085-04_moveit_config)/launch/moveit_rviz.launch">
    <arg name="rviz_config" value="$(find hebi_a-2085-04_moveit_config)/launch/moveit.rviz"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

</launch>
